motionplan/motionPlanner.go

```diff
@@ -483,12 +483,10 @@ func CheckPlan(
 	}
 
 	// create a list of segments to iterate through
-	var segments []*ik.Segment
+	segments := make([]*ik.Segment, 0, len(poses))
 	if relative {
-		segments = make([]*ik.Segment, 0, len(poses)+1)
-
 		// get the inputs we were partway through executing
-		checkFrameGoalInputs, err := sf.mapToSlice(plan.Trajectory()[0])
+		checkFrameGoalInputs, err := sf.mapToSlice(plan.Trajectory()[1])
```

Comment:
why is this changing?

Reply:
within `move_request.go/obstaclesIntersectPlan` the plan object passed into this function used to be `motionplan.RemainingPlan(plan, waypointIndex)`.
Now, to take advantage of the new `motionplan.Plan` object we do `motionplan.RemainingPlan(plan, waypointIndex-1)`.

Suppose we have constructed some plan with the following inputs `[ [0, 0, 0], [a, b, c], [d, e, f], [0, 0, 0] ]` and the following poses `[ startPose, middlePose, goalPose, goalPose ]`
Let our current input be `[a, b, x]` such that `x < c`. Hence, our wayPointIndex would be 1.

Therefore, `remainingPlan := motionplan.RemainingPlan(plan, waypointIndex-1)` gives us back the entirety of the plan. So the inputs we were part-way through would be `remainingPlan.Traj()[1]`.

The motivation for `waypointIndex-1` is that `remainingPlan.Path()[0]` is the `startPose` i.e. the `formerRunningPose`, which we can now gain without having to determine the `lastPose` and `pathPosition`

Reply:
It took some thinking through but I think this makes sense to me.  I think this makes sense to me.  I think the key insight is that we always are currently executing `Trajectory()[waypointIndex]` and the object being passed into this function includes the previous point as well.  I think some documentation explaining that the remainingPlan object actually includes the previous point the plan originates from would be valuable

Reply:
> I think some documentation explaining that the remainingPlan object actually includes the previous point the plan originates from would be valuable

good call out! I will update with a comment (both in move_request/obstaclesIntersectPlan and here)